src/command_modules/azure-cli-redis/azure/cli/command_modules/redis/_params.py

```diff
@@ -32,6 +32,19 @@ def __init__(self, value):
         dictval = shell_safe_json_parse(value)
         self.update(dictval)
 
+class VmSize(Enum): # pylint: disable=too-few-public-methods
```

Comment:
This is definitely one way of doing this but then it has the problem that you have to manually add things to this enum if the service supports something new.

I would recommend something like this:
```Python
family = size[0]
capacity = int(size[1:])
```

And then in the help for the param, you give examples of the format. "C1, P2, etc."  In this way, if the service starts to support additional combinations, then the CLI will automatically support them instead of artificially restricting them.